code/go/0chain.net/smartcontract/dbs/event/event.go

```diff
@@ -108,11 +60,50 @@ func (edb *EventDb) GetEvents(block int64) ([]Event, error) {
 }
 
 func (edb *EventDb) drop() error {
-	return edb.Store.Get().Migrator().DropTable(&Event{})
+	err := edb.Store.Get().Migrator().DropTable(&Event{})
+	if err != nil {
+		return err
+	}
+
+	err = edb.Store.Get().Migrator().DropTable(&Blobber{})
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-func (edb *EventDb) first() Event {
-	event := &Event{}
-	_ = edb.Store.Get().First(event)
-	return *event
+func (edb *EventDb) exists(event Event) (bool, error) {
+	var count int64
+	result := edb.Store.Get().
+		Model(&Event{}).
+		Where("tx_hash = ? AND index = ?", event.TxHash, event.Index).
+		Count(&count)
+	if result.Error != nil {
+		return false, fmt.Errorf("error counting events matching %v, error %v",
+			event, result.Error)
+	}
+	return count > 0, nil
+}
+
+func (edb *EventDb) removeDuplicate(events []Event) []Event {
+	for i := len(events) - 1; i >= 0; i-- {
+		exists, err := edb.exists(events[i])
+		if err != nil {
```

Comment:
Shall we not return the error here & stop the execution of function?

Reply:
No. Just because one of the events is bad does not mean that all the other events emitted this block should be forgotten. We don't want one bad transaction to damage other good transactions just because they happened in the same block.